pkg/diskmaker/discovery/sync_result.go

```diff
@@ -49,23 +49,14 @@ func (discovery *DeviceDiscovery) ensureDiscoveryResultCR() error {
 	nodeName := os.Getenv("MY_NODE_NAME")
 	namespace := os.Getenv("WATCH_NAMESPACE")
 	parentObjUID := os.Getenv("UID")
-	parentObjName := os.Getenv("POD_NAME")
+	parentObjName := os.Getenv("DISCOVERY_OBJECT_NAME")
 	if nodeName == "" || namespace == "" || parentObjUID == "" || parentObjName == "" {
```

Comment:
should we have a unit test for proper owner ref on discovery result objects?

Reply:
We do have some existing tests for that [here](https://github.com/openshift/local-storage-operator/blob/9c72cb708080ed6693dc090f52ed3282ca3fb6cc/pkg/diskmaker/discovery/sync_result_test.go#L144-L145) Will these suffice?